src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+    @Override
+    public int getDayOfWeek() {
+        if (day < 1) {
+            return 0;
+        }
+
+        return 1 + ((day - 1) % InternationalFixedChronology.DAYS_IN_WEEK);
+    }
+
+    private int getCalculatedDayOfWeek() {
+        return (day == 0) ? 7 : day == -1 ? 1 : 1 + (day - 1) % InternationalFixedChronology.DAYS_IN_WEEK;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    long getProlepticMonth() {
+        return getProlepticYear() * lengthOfYearInMonths() + getCalculatedMonth() - 1;
+    }
+
+    long getProlepticWeek() {
+        return ((long) prolepticYear) * InternationalFixedChronology.WEEKS_IN_YEAR +
+                getCalculatedMonth() * InternationalFixedChronology.WEEKS_IN_MONTH +
+                ((getCalculatedDayOfMonth() - 1) / InternationalFixedChronology.DAYS_IN_WEEK) - 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate minus(final TemporalAmount amount) {
+        return (InternationalFixedDate) amount.subtractFrom(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate minus(final long amountToSubtract, final TemporalUnit unit) {
+        return (amountToSubtract == Long.MIN_VALUE ? plus(Long.MAX_VALUE, unit).plus(1, unit) : plus(-amountToSubtract, unit));
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override  // for covariant return type
+    @SuppressWarnings ("unchecked")
+    public ChronoLocalDateTime<InternationalFixedDate> atTime(final LocalTime localTime) {
+        return (ChronoLocalDateTime<InternationalFixedDate>) ChronoLocalDate.super.atTime(localTime);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long until(final Temporal endExclusive, final TemporalUnit unit) {
+        return until(InternationalFixedDate.from(endExclusive), unit);
+    }
+
+    long until(final InternationalFixedDate end, final TemporalUnit unit) {
+        if (unit instanceof ChronoUnit) {
+            switch ((ChronoUnit) unit) {
+                case WEEKS:
+                    return weeksUntil(end);
+                case MONTHS:
+                    return monthsUntil(end);
+                default:
+                    break;
+            }
+        }
+
+        return super.until(end, unit);
+    }
+
+    /**
+     * Get the number of years from this date to the given day.
+     *
+     * @param end The end date.
+     * @return The number of years from this date to the given day.
+     */
+    private long yearsUntil(final InternationalFixedDate end) {
+        long startYear = getProlepticYear() * 512L + getDayOfYear();
+        long endYear = end.getProlepticYear() * 512L + end.getDayOfYear();
+
+        return (endYear - startYear) / 512L;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ChronoPeriod until(final ChronoLocalDate endDateExclusive) {
```

Comment:
Now I get it: your previous comment ("Remove yearsUntil(...) (unused, and incorrect anyways).") is an anticipation of the "piggy-back" you mention here.
Did I catch in a Yoda-moment, perhaps ? :-)
Changed the until method and got rid of yearsUntil.